mesonbuild/modules/pkgconfig.py

```diff
@@ -114,14 +122,8 @@ def _process_libs(self, libs, public):
                 # than needed build deps.
                 # See https://bugs.freedesktop.org/show_bug.cgi?id=105572
                 processed_libs.append(obj)
-                if public:
-                    if not hasattr(obj, 'generated_pc'):
-                        obj.generated_pc = self.name
```

Comment:
I think this is a breaking change. I believe it's a change in the right direction, but we should at least have a release note snippet that tell the users how to change their build definitions.

I'm not sure if there is a graceful way to deprecate this and only change over at a later point.

Reply:
Yes that's an intentional break, and indeed that must be added into release notes.

Reply:
Note, since it wasn't documented previously, and it's pretty corner case, I think it's not that bad.

Reply:
Release notes added.